packages/utils/src/useTableTools/FilterConfigBuilder/FilterConfigBuilder.js

```diff
@@ -0,0 +1,232 @@
+      acc[filterStateName] = state ? state : undefined;
+
+      return acc;
+    }, {});
+
+  categoryLabelForValue = (value) => {
+    const category = this.config.filter((category) => (category.items ? category.items.map((item) => item.value).includes(value) : false))[0];
+
+    return category ? category.label : value;
+  };
+
+  getCategoryForLabel = (query) => this.config.filter((item) => stringToId(item.key || item.label) === stringToId(query))[0] || {};
+
+  getItemByLabelOrValue = (query, category) => {
+    const categoryConfig = this.getCategoryForLabel(category);
+    const items =
+      categoryConfig.type !== conditionalFilterType.group
+        ? categoryConfig.items
+        : categoryConfig.items.flatMap((item) =>
+            item.items.map((subItem) => ({
+              ...subItem,
+              parentValue: item.value,
+            }))
+          );
+    const results = (items || []).filter((item) => item.value === query || item.label === query);
+
+    if (results.length === 1) {
+      return results[0];
+    } else if (results.length > 1) {
+      // eslint-disable-next-line no-console
+      console.info(`Multiple items found for ${query} in ${category}! Returning first one.`);
+      return results[0];
+    } else {
+      // eslint-disable-next-line no-console
+      console.info('No item found for ' + query + ' in ', category);
+    }
+  };
+
+  labelForValue = (value, category) => {
+    const item = this.getItemByLabelOrValue(value, category);
+    return item ? item.label : value;
+  };
+
+  valueForLabel = (label, category) => {
+    const item = this.getItemByLabelOrValue(label, category);
+    return item ? item.value : label;
+  };
+
+  applyFilterToObjectArray = (objects, activeFilters) => {
+    let filteredObjects = [...objects];
+
+    Object.keys(activeFilters).forEach((filter) => {
+      const category = this.getCategoryForLabel(filter);
+      const value = activeFilters[filter];
+      if (!category || !value) {
+        return;
+      }
+
+      if (value.length > 0 || (category.type === 'hidden' && typeof value === 'boolean')) {
+        filteredObjects = category.filter(filteredObjects, value);
+      }
+    });
+
+    return filteredObjects;
+  };
+
+  removeFilterFromFilterState = (currentState, filter) => (typeof currentState === 'string' ? '' : currentState.filter((value) => value !== filter));
+
+  removeFilterFromGroupFilterState = (currentState, filter, chipItem) => {
+    // eslint-disable-next-line
+        const { [chipItem.value]: _remove, ...newGroupState } = currentState[chipItem.parentValue];
+    return {
+      ...currentState,
+      [chipItem.parentValue]: newGroupState,
+    };
+  };
+
+  removeFilterWithChip = (chips, activeFilters) => {
+    const chipCategory = chips.category;
+    const chipLabel = chips.chips[0].name;
+    const chipItem = this.getItemByLabelOrValue(chipLabel, chipCategory);
+    const chipValue = chipItem ? chipItem.value : chipLabel;
```

Comment:
Let's use safe accessor in here
```suggestion
    const chipValue = chipItem?.value : chipLabel;
```